### PR TITLE
Notifications delta onboarding (EXPOSUREAPP-9615)

### PR DESCRIPTION
**Pull request by AlexanderAlferov**
## Changes
Delta onboarding for new notification management.

Screenshot 2021-09-29 at 13 04 01

## Story
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9497
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9615

## How to test
1. Go to testers menu
2. Onboarding options
3. Set version to previous (e.g. 2110000)
4. Scroll down to Notifications and Analytics part
5. Uncheck either only notifications or both
6. The correct delta onboarding order is: New features -> Notifications -> PPA
7. Check navigation from Notifications Delta Onboarding screen to application notification settings (OS level)

Note for testers: at this point it is impossible to get in position when notifications onboarding is complete and analytics is not, as analytics onboarding were introduced earlier (about Feb 2021).

### CODE REVIEW

File: Corona-Warn-App/src/main/res/values-de/notifications_strings.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    <!-- ####################################
+                  Delta Onboarding
+    ###################################### -->
+
+    <!-- XTXT: notification management: delta onboarding: body -->
+    <string name="nm_delta_onboarding_body">"Die Einstellungen für Mitteilungen werden ab jetzt einheitlich in den Geräteeinstellungen Ihres Smartphones verwaltet."</string>
+    <!-- XTXT: notification management: delta onboarding: subtitle -->
+    <string name="nm_delta_onboarding_subtitle">"Je nach Betriebssystem-Version und Gerätemodell können Sie festlegen, ob:"</string>
+    <!-- XBUT: notification management: delta onboarding: device settings button -->
+    <string name="nm_delta_onboarding_device_settings_button">"Geräteeinstellungen öffnen"</string>
```
Comment (InkenTownsend) [inline Corona-Warn-App/src/main/res/values-de/notifications_strings.xml:31]: @AlexanderAlferov Are the buttons in Android not always capitalized? GERÄTEEINSTELLUNGEN ÖFFNEN ? Sorry, I am not certain, I just saw that before.

Comment (AlexanderAlferov) [reply Corona-Warn-App/src/main/res/values-de/notifications_strings.xml:31]: @InkenTownsend , as we are using material design ([link](https://material.io/components/buttons#anatomy)), all buttons are capitalised. And you also do not need to worry about capitalisation in strings files, as it is done automatically. E.g. you can check the screenshot attached to this PR.

Comment (InkenTownsend) [reply Corona-Warn-App/src/main/res/values-de/notifications_strings.xml:31]: 👍 @AlexanderAlferov Thanks

**Review by InkenTownsend — DISMISSED** [review]
UA approved

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaonboardingFragment.kt [hunk 2]
```diff
@@ -60,6 +62,14 @@ class DeltaonboardingFragment : Fragment(R.layout.fragment_test_deltaonboarding)
...
         }
+
+        binding.switchNotificationsDeltaOnboarding.setOnCheckedChangeListener { _, value ->
+            viewModel.setNotificationsOnboardingDone(value)
+        }
+
+        binding.switchAnalyticsDeltaOnboarding.setOnCheckedChangeListener { _, value ->
+            viewModel.setAnalyticsOnboardingDone(value)
+        }
     }
```
Comment (axelherbstreith) [inline Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaonboardingFragment.kt:74]: very tiny remark: Please capitalize the first letter in "Onboarding" in the title to be consistent with other files

File: Corona-Warn-App/src/deviceForTesters/res/layout/fragment_test_deltaonboarding.xml [hunk 3]
```diff
@@ -200,5 +200,40 @@
...
+                android:layout_marginTop="@dimen/spacing_small"
+                android:text="Notifications onboarding finished"
+                app:layout_constraintTop_toBottomOf="@id/notifications_delta_onboarding_title" />
+
+            <com.google.android.material.switchmaterial.SwitchMaterial
+                android:id="@+id/switch_analytics_delta_onboarding"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="@dimen/spacing_small"
+                android:text="Analytics onboarding finished"
```
Comment (axelherbstreith) [inline Corona-Warn-App/src/deviceForTesters/res/layout/fragment_test_deltaonboarding.xml:233]: hardcoded? 👀

Comment (AlexanderAlferov) [reply Corona-Warn-App/src/deviceForTesters/res/layout/fragment_test_deltaonboarding.xml:233]: This is menu for testers - we always had hardcoded strings there :)

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaonboardingFragment.kt (hunk 2 above)
Comment (AlexanderAlferov) [reply Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaonboardingFragment.kt:74]: Good point: done ✅

File: Corona-Warn-App/src/deviceForTesters/res/layout/fragment_test_deltaonboarding.xml (hunk 3 above)
Comment (axelherbstreith) [reply Corona-Warn-App/src/deviceForTesters/res/layout/fragment_test_deltaonboarding.xml:233]: ah alright :)

**Review by InkenTownsend — DISMISSED** [review]
UA reviewed

**Review by axelherbstreith — APPROVED** [review]
lgtm! Onboarding order is maintained 💯

**Review by jurajkusnier — APPROVED** [review]

**Review by ChristinaTsigganos — APPROVED** [review]
UA approved

**Pull request merged**